nodeclient/http_api_client.go

```diff
@@ -321,6 +355,65 @@ func (client *Client) BlockIssuance(ctx context.Context) (*apimodels.IssuanceBlo
 	return res, nil
 }
 
+func (client *Client) Congestion(ctx context.Context, accountID iotago.AccountID) (*apimodels.CongestionResponse, error) {
+	res := new(apimodels.CongestionResponse)
+	query := fmt.Sprintf(RouteCongestion, hexutil.EncodeHex(accountID[:]))
+	//nolint:bodyclose
+	if _, err := client.Do(ctx, http.MethodGet, query, nil, res); err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func (client *Client) Rewards(ctx context.Context, outputID iotago.OutputID) (*apimodels.ManaRewardsResponse, error) {
+	res := &apimodels.ManaRewardsResponse{}
+	query := fmt.Sprintf(RouteRewards, hexutil.EncodeHex(outputID[:]))
+	//nolint:bodyclose
+	if _, err := client.Do(ctx, http.MethodGet, query, nil, res); err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func (client *Client) Staking(ctx context.Context) (*apimodels.AccountStakingListResponse, error) {
```

Comment:
We should also include an endpoint for Delegation

Reply:
oh But the delegation strictly for rewards? then /rewards endpoint covers both staking and delegation.
The staking endpoint was renamed to `validators` it just better suits the purpose which is for user to have a list of possible validators to delegate to.